app/models/host/managed.rb

```diff
@@ -306,29 +306,6 @@ def pm_fqdn
     (puppetmaster == "puppet") ? "puppet.#{domain.name}" : puppetmaster.to_s
   end
 
-  # Cleans Certificate and enable Autosign
-  # Called before a host is given their provisioning template
-  # Returns : Boolean status of the operation
-  def handle_ca
-    # If there's no puppetca, tell the caller that everything is ok
-    return true unless Setting[:manage_puppetca]
-    return true unless puppetca?
-
-    # From here out, we expect things to work and return true
-    return false unless respond_to?(:initialize_puppetca, true)
-    return false unless initialize_puppetca
-    return false unless delCertificate
```

Comment:
Do we need to queue this as well before we set the autosign entry? This is supposed to clean existing certs on the CA to make sure we start fresh. Do you mind adding a test that this is called?

Reply:
Will add this.